File: src/main/kotlin/ru/secondmemory/repository/InMemoryCardRepository.java
package ru.secondmemory.repository;

import ru.secondmemory.model.Card;

import java.util.Collection;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicInteger;

public class InMemoryCardRepository implements CardRepository {
    private final Map<Integer, Card> repository = new ConcurrentHashMap<>();
    private final AtomicInteger counter = new AtomicInteger(0);

//    {
//        CardsUtil.cards.forEach(this::save);
//    }

    @Override
    public Card save(Card card) {
//        if (card.isNew()) {
//            card.setId(counter.incrementAndGet());
//            repository.put(card.getId(), card);
        return card;
//        }
//        // handle case: update, but not present in storage
//        return repository.computeIfPresent(card.getId(), (id, oldCard) -> card);
    }

    @Override
    public boolean delete(int id) {
        return repository.remove(id) != null;
    }

    @Override
    public Card get(int id) {
        return repository.get(id);
    }

    @Override
    public Collection<Card> getAll() {
        return repository.values();
    }
}

File: src/main/kotlin/ru/secondmemory/repository/InMemoryCardRepository.kt
package ru.secondmemory.repository

import ru.secondmemory.model.Card
import ru.secondmemory.model.CardType
import ru.secondmemory.model.Cards
import ru.secondmemory.util.getEnumCardsTestData
import java.util.*

class InMemoryCardRepository : CardRepository {
    private val memoryData: EnumMap<CardType, Cards> = getEnumCardsTestData()

    override fun save(type: CardType, card: Card): Card? {
        memoryData[type]?.addCard(card.key, card)
        return get(type, card.key)
    }

    override fun delete(type: CardType, key: String): Boolean {
        memoryData[type]?.cards?.remove(key) ?: return false
        return true
    }

    override fun get(type: CardType, key: String): Card? {
        return memoryData[type]?.cards?.get(key) ?: return null
    }

    override fun getAllByType(type: CardType): List<Card> {
        return memoryData[type]!!.cards.values.toList()
    }
}